watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -272,41 +272,48 @@ def build(self):
         self.Liq = AqueousPhase()
 
         # list to hold all species (including water)
-        self.component_list = Set()
+        self.component_list = Set(dimen=1)
 
         # components
         self.H2O = Solvent()
 
-        # blank sets
-        self.cation_set = Set()
-        self.anion_set = Set()
-        self.solute_set = Set()
-        self.ion_set = Set()
+        # Other component sets
+        self.solute_set = Set(dimen=1)  # All components except Solvent() ("H2O")
+        self.cation_set = Set(dimen=1)  # Components with charge >0
+        self.anion_set = Set(dimen=1)  # Components with charge <0
+        self.neutral_set = Set(dimen=1)  # Components with charge =0
+        self.ion_set = Set(dimen=1)  # All Ion Components (cations + anions)
 
+        # Group components into different sets
         for j in self.config.solute_list:
+            if j == "H2O":
+                raise ConfigurationError(
+                    "'H2O'is reserved as the default solvent and cannot be a solute."
+                )
+            self.add_component(j, Solute())
             if j in self.config.charge:
                 if self.config.charge[j] == 0:
                     raise ConfigurationError(
                         "The charge property should not be assigned to the neutral component: {}".format(
                             j
                         )
                     )
-                elif self.config.charge[j] > 0:
+                if self.config.charge[j] > 0:
```

Comment:
```suggestion
                elif self.config.charge[j] > 0:
```
The logic was clearer with this as an elif. Alternatively, we might have `self.config.charge[j] > 0`, `self.config.charge[j] < 0` else raise the configuration error.

Reply:
@bknueven , I double-checked this way and it doesn't make it work (actually having a fail in building the component_list).  I think the trick is in this file: https://github.com/IDAES/idaes-pse/blob/main/idaes/core/base/components.py . I see two reasons for me having to add j in `Solute()` before going to Cation and Anion: 1) this can trigger it being added into the `component_list` by calling a `_add_to_component_list` function, 2) this can let `Anion()` and `Cation()` also registered in `Solute()`.  The current main version actually uses the logic you wrote but it exclude the ions out of `Solute()`.  Let me know if you find I mistake anything.  Thanks.

Reply:
@andrewlee94 are we doing something wrong here? It seems strange that the API would require us to add a component just to delete it, in order to get some side effect.

Even if we're not using the IDAES API as intended, I wonder if we should implement a different work-around. My preference would be to directly code-in what we want to happen as opposed to relying on the side effect of `add_component` followed by calling `del_component`, which to be just looks like a silly refactor away.

@lbibl if we do decide to leave this as-is, it needs a comment in the code similar to what you say in your commment on GitHub.

Reply:
Commented notes for this are added.  

One obvious profit of using IDAES components is to have the `is_solute()` fun working for these sets, although `is_solute()` can be simply replace by `in solute_set` wherever needed. 

At this time I would recommend as-is because 1) the "silly refactor" doesn't really cost anything and 2) of saving time.

Reply:
Another (best in my view) resolution is to make changes in https://github.com/IDAES/idaes-pse/blob/main/idaes/core/base/components.py  to let `Cation()` and `Anion()`, and maybe `Ion()` too, possess methods to add their members into the corresponding sets of their parent classes  (Solute for all, Ion for the other two).  This could be @andrewlee94 's call.

Reply:
> At this time I would recommend as-is because 1) the "silly refactor" doesn't really cost anything and 2) of saving time.

I wasn't being clear: when I look at this code my first thought is, "why do we add a component only to delete it"? So we need comments (which you've added) to explain that. Otherwise somebody else will come along with the same thought in this comment and it won't work.

I don't really like things as they are, but if we're using IDAES correctly I can live with it.

Reply:
@bknueven @lbibl As far as I can tell (and I am having a little trouble following everything here), the issue is that you are using the IDAES `Component` objects, but don;t agree with/want their definition of the component sets. so in short, if you don;t want the IDAES set definitions you should not be using the IDAES `Component` objects.

Of course, this would mean you lose the functionality of the `Component` objects (or would need to replicate this), but that is the trade off you would need in this case.

Reply:
@lbibl is the definition of component the same throughout WaterTAP, and hence divergent from the IDAES definitions?

Reply:
@bknueven I am fairly certain most of the WaterTap property packages use the standard IDAES `Components`, and thus component sets - for one the eNRTL implementation uses the IDAES modular properties (which is what the `Components` were originally written for).

Reply:
> @bknueven @lbibl As far as I can tell (and I am having a little trouble following everything here), the issue is that you are using the IDAES `Component` objects, but don't agree with/want their definition of the component sets. So in short, if you don't want the IDAES set definitions you should not be using the IDAES `Component` objects.
> 
> Of course, this would mean you lose the functionality of the `Component` objects (or would need to replicate this), but that is the trade off you would need in this case.

If we put this PR aside for a moment,  I saw an internal inconsistency in IDAES's components.py.  `Cation` and `Anion` there inherited `Solute` and running `is_solute()` on Cation or Anion member returns True.  But the two subclasses did not add their members in the `solute_set` nor the component_list .

Reply:
> @lbibl is the definition of component the same throughout WaterTAP, and hence divergent from the IDAES definitions?

I didn't originally write the original version of this property package, neither the rest of property packages in Watertap.  I can only answer this PR does not change the definition of component from its original version of this property package.  I thought it's consistent (I could be wrong) to IDAES, because the components.py has been used from the first in this specific property package,  but always with some band-aid to walk around the inconsistency I described above.

Reply:
@lbibl The problem is that your definition is fundamentally inconsistent with the one that IDAES uses - that is why there is an issue. It should be noted the "solute_set" in IDAES was always intended to be "non-ionic solutes" but we could never think of a good name for it - the reason that ions are not added to `solute_set` is so that we can clear identify sub-types.

further, the ionic `Components` were only intended to be used in limited cases where the property package would decide what to do with them (hence why there are not added to `solute_set` or `component_list`). Basically, they were not designed to be used the way you are (for better or worse).

Reply:
Hi @andrewlee94 , thanks for your clarification.  I heard all the message.  Just to repeat one thing (to make sure you didn't miss):  running `is_solute()` on `Ion`, `Cation`  and `Anion` returns `True` although they are not in `solute_set`.   I think this could cause confusion in mistakes for anyone using this file.

Reply:
Yes - that is correct, and is because of the naming of `solute_set` - as I said it really should have a different name to indicate that it is "non-ionic solutes".

Reply:
To make my conclusion over this discussion, 1) I myself recommend keeping this PR as-is because it functions as we planned; 2) with all information shared,  I'll follow reviewers' recommendation/decision on alternative fate of this PR .

Reply:
I'm not opposed to adopting our own definition of these sets in principle, if it's what makes sense for WaterTAP. I would also be happy to mock up our own components which deviate from those in IDAES.

I do think that having different definitions is different property packages will ultimately lead to confusion. WaterTAP should at least be internally consistent, and ideally consistent with IDAES.

Reply:
I approved the PR but I do share @bknueven's concerns

Reply:
It seems to me that a corresponding modification should be considered in IDAES' generic property framework (forgetting what the newest name for it is) to maintain consistency where possible, if it wouldn't break a million things in IDAES. For instance, renaming solute_set as neutral_set and then defining solute_set to include both ions and neutral solutes--maybe @lbibl can post an issue in IDAES regarding that. Thoughts?

Reply:
I am not sure how easy it would be to change things on the IDAES side, plus IDAES will not have the resources to do this anytime soon (but it is open-source so WaterTap can always propose a fix if it is important enough).

However, "neutral_set" is probably not a great name, as that could include solvents as well (which are generally neutral). Also, simpler models may wish to declare ions as general solutes in case where they don't need more complex aqueous phase behaviours.

Reply:
My honest thought is time has been overly used on a simple matter.  With all considered, my recommendation at this time is to either keep things as they are, (as far as I see, this PR does not cause inconsistency in WaterTap but only make things clearer and easier) , or write a WaterTap Component file (which is fairly short) to be consistent with this PR.

Reply:
I don't think the key is about naming- but putting ions in solute in components. I don't see a valid reason to let ions give `Is_solute() == True` but `in solute_set == False`.